### PR TITLE
Fixed gong script for main v2

### PR DESCRIPTION
**Pull request by f7deleon**
Fixed gong script for main v2

## :warning: Warnings:
i added the following line it don't do anything but just in case what do you think? i can remove it

```
   sed -i "s/$BASE_PROJECT_NAME/$REAL_PROJECT_NAME/" proguard-rules.pro
```

### CODE REVIEW

File: gong_setup.sh [hunk 1]
```diff
@@ -29,25 +29,32 @@ function changeProjectName() {
   new_path=$(sed "s/\./\//g" <<<"$PACKAGE_NAME")
   gong_new_path=$(sed "s/\./\//g" <<<"$BASE_PROJECT_PAKAGE_NAME")
   first_gong_folder=$(sed 's/\..*//' <<<"$BASE_PROJECT_PAKAGE_NAME")
-  cd "app/src/" || exit 1
+
+  cd "app/" || exit 1
+  sed -i "s/$BASE_PROJECT_PAKAGE_NAME/$PACKAGE_NAME/" proguard-rules.pro
+  sed -i "s/$BASE_PROJECT_NAME/$REAL_PROJECT_NAME/" proguard-rules.pro
```
Comment (mirland) [inline gong_setup.sh:35]: I'f replate in all proguard files, I'd edit line L26

```
find . -type f \( -name "*.xml" -o -name "*.gradle" -o -name "*.kt" -o -name "*.java" -o -name "*.pro" \) -exec perl -i -pe "s/$BASE_PROJECT_PAKAGE_NAME/$PACKAGE_NAME/g" {} \;
 
```

File: gong_setup.sh [hunk 2]
```diff
@@ -29,25 +29,32 @@ function changeProjectName() {
...
+
+  cd "src/" || exit 1
 
   if [ -d "$TEMPORAL_FOLDER" ]; then rm -Rf $TEMPORAL_FOLDER; fi
   mkdir "$TEMPORAL_FOLDER"
+  
+  for entry in $fileNames 
+  do
```
Comment (mirland) [inline gong_setup.sh:43]: i'd join it to be consistent with the if sentences
```suggestion
  for entry in $fileNames ;  do
```

Comment (mirland) [reply gong_setup.sh:43]: where `fileNames` is defined?

File: gong_setup.sh [hunk 3]
```diff
@@ -29,25 +29,32 @@ function changeProjectName() {
...
   mv $BASE_PROJECT_NAME "$PROJECT_NAME"
   cd "$PROJECT_NAME" || exit 1
 }
 
 function cloneAndSetupRepository() {
-  git clone --depth=1 $GIT_BASE_PROJECT_URL --quiet
+  git clone $GIT_BASE_PROJECT_URL --quiet
   cd "$BASE_PROJECT_NAME" || exit 1
+  git checkout main-v2 --quiet
+  git branch --quiet | grep -v "main-v2" | xargs git branch -D --quiet
```
Comment (mirland) [inline gong_setup.sh:57]: I'd not harcode it, I'd use it from an optional args

File: gong_setup.sh (hunk 2 above)
Comment (f7deleon) [reply gong_setup.sh:43]: weird i thought i do it different

File: gong_setup.sh (hunk 1 above)
Comment (f7deleon) [reply gong_setup.sh:35]: nice

File: gong_setup.sh [hunk 4]
```diff
@@ -3,6 +3,7 @@
 BASE_PROJECT_PAKAGE_NAME="com.xmartlabs.gong"
 BASE_PROJECT_NAME="gong"
 GIT_BASE_PROJECT_URL="https://github.com/xmartlabs/gong.git"
+GIT_BRANCH="${BASH_ARGV[0]}"
```
Comment (mirland) [inline gong_setup.sh:6]: cannot use $1?

**Review by mirland — APPROVED** [review]

**Review by mirland — CHANGES_REQUESTED** [review]
it seems that it's not worling 
```bash
bash <(curl -s https://raw.githubusercontent.com/xmartlabs/gong/261adae53a2e3df95116a562fe80c4b8c2d30d0b/gong_setup.sh) main-v2
```

**Review by mirland — APPROVED** [review]
I tested again and it's working! my bad

**Pull request merged**